File: spring-security/src/main/java/com/example/springsecex/runner/DefaultApplicationRunner.java
package com.example.springsecex.runner;

import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.core.annotation.Order;
import org.springframework.stereotype.Component;

import lombok.extern.slf4j.Slf4j;

@Slf4j
@Component
@Order(2)
public class DefaultApplicationRunner implements ApplicationRunner {
    /*
     * $ gradle bootRun --args='--opt-arg1=value1 --opt-arg2=value2 --non-opt-arg1 0123 non-opt-arg2 -s'
     * $ java -jar executable.jar --opt-arg1=value1 --opt-arg2=value2 --non-opt-arg1 0123 non-opt-arg2 -s
     * $ java -cp non-executable.jar <main-class-name> --opt-arg1=value1 --opt-arg2=value2 --non-opt-arg1 0123 non-opt-arg2 -s
     */
    @Override
    public void run(ApplicationArguments args) throws Exception {
        var optNames = args.getOptionNames();
        for (var name : optNames) {
            log.info("option: name={}, value={}", name, args.getOptionValues(name));
        }

        var nonOptArgs = args.getNonOptionArgs();
        for (var arg : nonOptArgs) {
            log.info("non option args: {}", arg);
        }

        var srcArgs = args.getSourceArgs();
        for (var arg : srcArgs) {
            log.info("source arg: {}", arg);
        }
    }
}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/runner/DefaultApplicationRunner.kt
package com.example.springwebex.runner

import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component
import io.github.oshai.kotlinlogging.KotlinLogging
@Component
@Order(2)
class DefaultApplicationRunner : ApplicationRunner {
    private val log = KotlinLogging.logger {}
    override fun run(args: ApplicationArguments?) {
        log.info { "DefaultApplicationRunner: Application started" }
        
        if (args != null) {
            log.info { "Option Names: ${args.optionNames}" }
            log.info { "Non Option Args: ${args.nonOptionArgs}" }

            for (optionName in args.optionNames) {
                log.info { "Option - $optionName: ${args.getOptionValues(optionName)}" }
            }
        }
    }
}
